samza-api/src/main/java/org/apache/samza/sql/SamzaSqlRelRecord.java

```diff
@@ -0,0 +1,111 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one
+* or more contributor license agreements.  See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership.  The ASF licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License.  You may obtain a copy of the License at
+*
+*   http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package org.apache.samza.sql;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import org.codehaus.jackson.annotate.JsonProperty;
+
+
+/**
+ * Samza sql relational record. A record consists of list of column values and the associated column names.
+ * A column value could be nested, meaning, it could be another SamzaSqlRelRecord.
+ * Right now we do not store any metadata (like nullability, etc) other than the column name in the SamzaSqlRelRecord.
+ */
+public class SamzaSqlRelRecord implements Serializable {
+
+  @JsonProperty("fieldNames")
+  private final List<String> fieldNames;
```

Comment:
Since this class is implementing Serializable, it is better to force the type of List here to a concrete serializable implementation of List (e.g. ArrayList or LinkedList).

Reply:
Makes sense.